pallets/collator-selection/src/benchmarking.rs

```diff
@@ -107,22 +151,32 @@ benchmarks! {
 
 		<CandidacyBond<T>>::put(T::Currency::minimum_balance());
 		<DesiredCandidates<T>>::put(c + 1);
+
+		register_validators::<T>(c);
 		register_candidates::<T>(c);
 
 		let caller: T::AccountId = whitelisted_caller();
 		let bond: BalanceOf<T> = T::Currency::minimum_balance() * 2u32.into();
 		T::Currency::make_free_balance_be(&caller, bond.clone());
 
+		<session::Module<T>>::set_keys(
+			RawOrigin::Signed(caller.clone()).into(),
+			keys::<T>(c + 1),
+			vec![]
+		).unwrap();
+
 	}: _(RawOrigin::Signed(caller.clone()))
 	verify {
 		assert_last_event::<T>(Event::CandidateAdded(caller, bond / 2u32.into()).into());
 	}
 
 	// worse case is the last candidate leaving.
 	leave_intent {
-		let c in 1 .. T::MaxCandidates::get();
+		let c in (T::MinCandidates::get() + 1) .. T::MaxCandidates::get();
```

Comment:
```suggestion
		let c in T::MinCandidates::get().max(1) .. T::MaxCandidates::get();
```
Could it be that you want to address the possibility of `MinCandidates` being 0?

Reply:
No, what I want is to make sure that there is at least an extra candidate which is able to leave. If the number of candidates is not bigger than `MinCandidates` it will panic.